core/src/main/java/com/linecorp/armeria/common/HttpResponse.java

```diff
@@ -825,6 +826,30 @@ default HttpResponse mapError(Function<? super Throwable, ? extends Throwable> f
         return of(HttpMessage.super.mapError(function));
     }
 
+    /**
+     * Applies specified {@link Consumer} to the non-informational {@link ResponseHeaders}
+     * emitted by this {@link HttpResponse}.
+     *
+     * <p>For example:<pre>{@code
+     * HttpResponse response = HttpResponse.of(ResponseHeaders.of(HttpStatus.OK));
+     * HttpResponse result = response.peekHeaders(headers -> {
+     *      assert headers.status().equals(HttpStatus.OK);
+     * });
+     * }</pre>
+     */
+    default HttpResponse peekHeaders(Consumer<? super ResponseHeaders> action) {
+        requireNonNull(action, "action");
+        final StreamMessage<HttpObject> stream = peek(obj -> {
+            if (obj instanceof ResponseHeaders) {
+                final ResponseHeaders headers = (ResponseHeaders) obj;
+                if (!headers.status().isInformational()) {
+                    action.accept(headers);
+                }
+            }
+        });
```

Comment:
How about adding `StreamMessage.peek(Consumer<? super T>, Class<? extends T> type)` and implementing this using the new method?